### PR TITLE
Improve makefile

### PR DESCRIPTION
**Pull request by arnolanglade**
### How Makefile works
A simple makefile consists of "rules" with the following shape:

```
target ... : prerequisites ...
        command
        ...
        ...
```

A **target** is usually the name of a file that is generated by a program; examples of targets are executable or object files. A target can also be the name of an action to carry out, such as `clean' (see section Phony Targets).

A **prerequisite** is a file that is used as input to create the target. A target often depends on several files.

A **command** is an action that make carries out. A rule may have more than one command, each on its own line. Please note: you need to put a tab character at the beginning of every command line! This is an obscurity that catches the unwary. 

Extra information [here](https://ftp.gnu.org/old-gnu/Manuals/make-3.79.1/html_chapter/make_2.html)

### What did I need to do have working PIM before the Makefile?
```bash
$ docker-compose up
$ bin/docker/pim-dependencies.sh
$ bin/docker/pim-setup.sh
$ bin/docker/pim-initialize.sh
```

### What do I need to do have working PIM after the Makefile 
```
make up
make init-pim
```

Makefile is more intelligent then a our script. For instance the target `up` depends on the target `vendor`. Make will install backend dependencies automatically if this directory does not exist. That means when I run this command I sure that my pim will be well setup because this target depend on every thing it needs.

We only need to run 2 commands. The target `up` may depend on `init-pim` but it means every time you run `make up` the database and run webpack.

### Note
I don't delete docker shell scripts so far. Moreover this makefile are the bare minimum to run a PIM, we need to improve it. Makefile will automatically include all file in make-file/*.mk. For now, I have added a file per bounded context with a single target (coupling rule). Each team should add its custom targets or you can add you custom targets. **Caution:** those targets are **optimized for docker**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,3 +38,66 @@ enrichment-coupling:
...
+
+node_modules: package.json
+	$(YARN_EXEC) install
+
+behat.yml:
+	cp ./behat.yml.dist ./behat.yml
+	sed -i "s/127.0.0.1\//httpd-behat\//g" ./behat.yml
+	sed -i "s/127.0.0.1/selenium/g" ./behat.yml
+
+app/config/parameters.yml:
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply Makefile]: This is another way to do it but the best way is to use env vars. For now, I have used the command from the pim-setup.sh

File: Makefile [hunk 2]
```diff
@@ -1,8 +1,11 @@
 DOCKER ?= false
-PHP_EXEC :=php
+PHP_EXEC = php
+YARN_EXEC = yarn
 
 ifeq ($(DOCKER),true)
+	DOCKER_COMPOSE = docker-compose
```
Comment (anaelChardan) [inline Makefile]: Should be outside of the if as it is an invariant

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (anaelChardan) [reply Makefile]: This target works only for docker, am I right ?

File: Makefile [hunk 3]
```diff
@@ -35,3 +38,66 @@ enrichment-coupling:
...
+	sed -i "s/127.0.0.1\//httpd-behat\//g" ./behat.yml
+	sed -i "s/127.0.0.1/selenium/g" ./behat.yml
+
+app/config/parameters.yml:
+	cp ./app/config/parameters.yml.dist ./app/config/parameters.yml
+	# Sed commands should be removed when env var will be introduce in the PIM
+	sed -i "s/database_host:.*localhost/database_host: mysql/g" ./app/config/parameters.yml
+	sed -i "s/localhost: 9200/elastic:changeme@elasticsearch:9200/g" ./app/config/parameters.yml
+
+app/config/parameters_test.yml:
```
Comment (anaelChardan) [inline Makefile]: Same here

File: Makefile [hunk 4]
```diff
@@ -35,3 +38,66 @@ enrichment-coupling:
...
+	sed -i "s/product_and_product_model_index_name:.*akeneo_pim_product_and_product_model/product_and_product_model_index_name:  test_akeneo_pim_product_and_product_model/g" ./app/config/parameters_test.yml
+	sed -i "s/record_index_name:.*akeneo_referenceentity_record/record_index_name:                     test_akeneo_referenceentity_record/g" ./app/config/parameters_test.yml
+	sed -i "s/product_proposal_index_name:.*akeneo_pim_product_proposal/product_proposal_index_name:           test_akeneo_pim_product_proposal/g" ./app/config/parameters_test.yml
+	sed -i "s/published_product_index_name:.*akeneo_pim_published_product/published_product_index_name:          test_akeneo_pim_published_product/g" ./app/config/parameters_test.yml
+	sed -i "s/published_product_and_product_model_index_name:.*akeneo_pim_published_product_and_product_model/published_product_and_product_model_index_name: test_akeneo_pim_published_product_and_product_model/g" ./app/config/parameters_test.yml
+
+docker-compose.override.yml:
+	cp docker-compose.override.yml.dist docker-compose.override.yml
```
Comment (anaelChardan) [inline Makefile]: I would add the docker assertion here

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (arnolanglade) [reply Makefile]: yes, we may define some var make it more flexible.

File: Makefile (hunk 4 above)
Comment (arnolanglade) [reply Makefile]: this target only depend on `up`, we don't need to add an assertion. We need to keep it simple.

File: Makefile [hunk 5]
```diff
@@ -1,37 +1,90 @@
...
 .DEFAULT_GOAL := help
 
 .PHONY: help
 help:
 	@echo ""
-	@echo "Akeneo Pim Community Dev available targets:"
+	@echo "Caution: those targets are optimized for docker"
 	@echo ""
-	@echo "If you want to use docker, use an environment variable DOCKER=true"
+	@echo "Please add your custom Makefile in the dicretory "make-file". They will be automotically loaded!"
```
Comment (anaelChardan) [inline Makefile]:
```suggestion
	@echo "Please add your custom Makefile in the directory "make-file". They will be automatically loaded!"
```

**Pull request merged**